core/templates/pages/contributor-dashboard-page/services/contribution-and-review.service.ts

```diff
@@ -16,158 +16,168 @@
-    var _UPDATE_TRANSLATION_HANDLER_URL = (
-      '/updatetranslationsuggestionhandler/<suggestion_id>'
-    );
-    var _UPDATE_QUESTION_HANDLER_URL = (
-      '/updatequestionsuggestionhandler/<suggestion_id>'
-    );
+import { downgradeInjectable } from '@angular/upgrade/static';
+import { Injectable } from '@angular/core';
+import { AppConstants } from 'app.constants';
+import { ContributionAndReviewBackendApiService }
+  from './contribution-and-review-backend-api.service';
+import { SuggestionBackendDict } from 'domain/suggestion/suggestion.model';
+import { StateBackendDict } from 'domain/state/StateObjectFactory';
+import { ImagesData } from 'services/image-local-storage.service';
+
+export interface OpportunityDict {
+  'skill_id': string;
+  'skill_description': string;
+}
+
+export interface FetchSuggestionsResponse {
+  'target_id_to_opportunity_dict': {
+    [targetId: string]: OpportunityDict;
+  };
+  suggestions: SuggestionBackendDict[];
+}
 
-    var _fetchSuggestionsAsync = async function(url) {
-      return $http.get(url).then(function(res) {
-        var suggestionIdToSuggestions = {};
-        var targetIdToDetails = res.data.target_id_to_opportunity_dict;
-        res.data.suggestions.forEach(function(suggestion) {
+interface FetchSuggestions {
+  [targetId: string]: {
+    suggestion: SuggestionBackendDict;
+    details: OpportunityDict;
+  };
+}
+
+@Injectable({
+  providedIn: 'root',
+})
+export class ContributionAndReviewService {
+  constructor(
+    private contributionAndReviewBackendApiService:
+      ContributionAndReviewBackendApiService
+  ) {}
+
+  private async _fetchSuggestionsAsync(
+      fetchType: string
+  ): Promise<FetchSuggestions> {
+    return (
+      this.contributionAndReviewBackendApiService.fetchSuggestionsAsync(
+        fetchType
+      ).then((data) => {
+        let suggestionIdToSuggestions: FetchSuggestions = {};
+        let targetIdToDetails = data.target_id_to_opportunity_dict;
+        data.suggestions.forEach((suggestion) => {
           suggestionIdToSuggestions[suggestion.suggestion_id] = {
             suggestion: suggestion,
             details: targetIdToDetails[suggestion.target_id]
           };
         });
         return suggestionIdToSuggestions;
+      })
+    );
+  }
+
+  async getUserCreatedQuestionSuggestionsAsync(): Promise<FetchSuggestions> {
+    return this._fetchSuggestionsAsync('SUBMITTED_QUESTION_SUGGESTIONS');
+  }
+
+  async getReviewableQuestionSuggestionsAsync(): Promise<FetchSuggestions> {
+    return this._fetchSuggestionsAsync('REVIEWABLE_QUESTION_SUGGESTIONS');
+  }
+
+  async getUserCreatedTranslationSuggestionsAsync(): Promise<FetchSuggestions> {
+    return this._fetchSuggestionsAsync('SUBMITTED_TRANSLATION_SUGGESTIONS');
+  }
+
+  async getReviewableTranslationSuggestionsAsync(): Promise<FetchSuggestions> {
+    return this._fetchSuggestionsAsync('REVIEWABLE_TRANSLATION_SUGGESTIONS');
+  }
+
+  resolveSuggestionToExploration(
```

Comment:
Same naming comments apply in this file as well.

Reply:
done